lib/ansible/playbook/conditional.py

```diff
@@ -212,17 +213,32 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
             # to be looking for an undefined variable, return True, otherwise fail
             try:
                 # first we extract the variable name from the error message
-                var_name = re.compile(r"'(hostvars\[.+\]|[\w_]+)' is undefined").search(str(e)).groups()[0]
+                search_var = re.compile(r"'(hostvars\[.+\]|[\w_]+)' is undefined").search(to_text(e))
+                if not search_var and (re.compile(r"ansible.parsing.yaml.objects.AnsibleUnicode object has no element Undefined").search(to_text(e)) or
+                                       "list object has no element" in to_text(e)):
```

Comment:
We don't have to use a regex for looking for ```ansible.prsing.yaml.objects.AnsibleUnicode``` since it's just a substring search (no wildcards, character classes, or other regex features).  Something like this would be better:
``` python
if not search_var and ("ansible.parsing.yaml.objects.AnsibleUnicode object has no element Undefined" in to_text(e)) or "list object has no element" in to_text(e)):
```